main.go

```diff
@@ -2,47 +2,56 @@ package main
 
 import (
 	"flag"
-	"github.com/golang/glog"
-	"github.com/mchandramouli/haystack-kube-sidecar-injector/httpd"
-	"github.com/mchandramouli/haystack-kube-sidecar-injector/routes"
 	"os"
 	"os/signal"
 	"syscall"
-	"time"
+
+	"github.com/golang/glog"
+	"github.com/mchandramouli/haystack-kube-sidecar-injector/httpd"
+	"github.com/mchandramouli/haystack-kube-sidecar-injector/routes"
 )
 
 type config struct {
-	httpdConf httpd.Conf
+	httpdConf         httpd.Conf
 	sideCarConfigFile string
 }
 
 func main() {
 	conf := readConfig()
-
 	simpleServer := httpd.NewSimpleServer(conf.httpdConf)
 
 	var err error
-	if err = addRoutes(simpleServer, conf); err == nil {
-		if err = startHttpsServer(simpleServer); err == nil {
-			glog.Infof("SimpleServer listening in port %v", simpleServer.Port())
-			wait(func() {
-				glog.Infof("Shutting down initiated")
-				simpleServer.Shutdown()
-			})
-			return
+	defer func() {
```

Comment:
This simple sugar allows us to do early termination instead of nested ifs.

Reply:
That is neat. Thanks